hplot/s2d.go

```diff
@@ -82,18 +86,45 @@ func (pts *S2D) withBand() error {
 	}
 
 	var (
+		top plotter.XYs
+		bot plotter.XYs
+	)
+
+	switch pts.Steps {
+
+	case NoSteps:
 		top = make(plotter.XYs, pts.Data.Len())
 		bot = make(plotter.XYs, pts.Data.Len())
-	)
-	for i := range top {
-		x, y := pts.Data.XY(i)
-		ymin, ymax := yerr.YError(i)
-		top[i].X = x
-		top[i].Y = y + math.Abs(ymax)
-		bot[i].X = x
-		bot[i].Y = y - math.Abs(ymin)
-	}
+		for i := range top {
+			x, y := pts.Data.XY(i)
+			ymin, ymax := yerr.YError(i)
+			top[i].X = x
+			top[i].Y = y + math.Abs(ymax)
+			bot[i].X = x
+			bot[i].Y = y - math.Abs(ymin)
+		}
 
+	case HiSteps:
+		top = make(plotter.XYs, 2*pts.Data.Len())
+		bot = make(plotter.XYs, 2*pts.Data.Len())
+		xerr := pts.Data.(plotter.XErrorer)
+		for i := range top {
+			idata := i / 2
+			x, y := pts.Data.XY(idata)
+			xmin, xmax := xerr.XError(idata)
+			ymin, ymax := yerr.YError(idata)
+			top[i].X = x - math.Abs(xmin)
```

Comment:
perhaps use a `switch` instead of setting those fields twice in the case of `i%2 != 0`?
_e.g.:_
```go
switch {
case i%2 != 0:
     top[i].X = x + math.Abs(xmax)
     // ...
default:
     top[i].X = x - math.Abs(xmin)
    // ...
}
```